### PR TITLE
Added Updating Schedule

### PR DESCRIPTION
**Pull request by pcockrell6**

### CODE REVIEW

File: API/routes/api/make_schedule.js [hunk 1]
```diff
@@ -110,6 +107,96 @@ router.route('/')
...
+
+    if (request.body.startTime) newAttr.startTime = request.body.startTime;
+    if (request.body.endTime) newAttr.endTime = request.body.endTime;
+    if (request.body.isOneTime) newAttr.isOneTime = request.body.isOneTime;
+    if (request.body.classDate) newAttr.classDate = request.body.classDate;
+    if (request.body.dayOftheWeek) newAttr.dayOftheWeek = request.body.dayOftheWeek;
+
+    /* Find the class in the DB */
+
+    classModel.find({"className": request.body.className}, 'class starts ends one-time day date', (e, query) => {
```
Comment (zachsanchez113) [inline API/routes/api/make_schedule.js:165]: Is `one-time` the attribute in the DB? I haven't checked, but make sure that the attributes you're trying to fetch are spelled correctly and have the right names, as that's what the DB will return, and if it's empty, nothing will come back

File: API/routes/api/make_schedule.js [hunk 2]
```diff
@@ -110,6 +107,96 @@ router.route('/')
...
+      if (!query[0] || query[0] == "") return next(createError(400, "No class exists to update"));
+
+      if (process.env.NODE_ENV == "DEBUG") {
+        
+        console.log("The list of new attributes: " + JSON.stringify(newAttr));
+        console.log("What was actually returned: " + JSON.stringify(query));
+
+      }
+
+      if (!query || query == "") return next(createError(400, "No class specified to update"));
```
Comment (zachsanchez113) [inline API/routes/api/make_schedule.js:178]: Might want the error to say something like `That class doesn't exist`.

File: API/routes/api/make_schedule.js [hunk 3]
```diff
@@ -110,6 +107,96 @@ router.route('/')
...
+        "startTime": query.startTime,
+        "endTime": query.endTime,
+        "isOneTime": query.isOneTime,
+        "dayOfTheWeek": query.dayOftheWeek,
+        "classDate": query.classDate
+
+      };
+
+      console.log("Class update success for class " + request.body.className + ".");
+      return response.statust.(200).send(updatedClass);
```
Comment (zachsanchez113) [inline API/routes/api/make_schedule.js:194]: Typo

File: API/routes/api/make_schedule.js (hunk 1 above)
Comment (zachsanchez113) [inline API/routes/api/make_schedule.js:165]: This doesn't update the DB, all it does is fetch the class. If the user comes back later, the class won't be altered. For that you need the `findOneAndUpdate()` method: https://mongoosejs.com/docs/api.html#model_Model.findOneAndUpdate

Look at `profile.js` or `admin.js` for examples in the codebase that use this method.

**Comment by pcockrell6** [comment]
Fixed above issues. Hopefully, there aren't any big issues with this new commit

File: API/lib/schedule_db.js [hunk 4]
```diff
@@ -65,12 +65,16 @@ var ClassSchema = new Schema({
...
     required: false
     
   }
 
-});
+}, {timestamps: true});
+
+/* Automatically expires after 30 days of being in the DB */
+
+Class.schema.index({createdAt: 1},{expireAfterSeconds: 2592000});
```
Comment (zachsanchez113) [inline API/lib/schedule_db.js:77]: This broke things, it's fixed in the codebase so I think I'll just include this in a PR for updating `files.js`

File: API/routes/api/get_schedule.js [hunk 5]
```diff
@@ -89,4 +89,50 @@ router.route('/')
...
   });
 
+  /* Find classes within a date range
+   * @param {Date} startDate - starting date of the range
+   * @param {Date} endDate - ending date of the range
+   * @return {undefined} 200 if successful
+   * @return {undefined} 400 if no date range was provided
+   */
+router.route('/')
+  .post(async (request, response, next) => {
```
Comment (zachsanchez113) [inline API/routes/api/get_schedule.js:99]: You can't do this twice for a single route (you did it for the root route `/`)

File: API/routes/api/make_schedule.js [hunk 6]
```diff
@@ -90,7 +87,7 @@ router.route('/')
           "endTime": request.body.endTime,
           "dayOftheWeek": (request.body.dayOftheWeek == null) ? "" : request.body.dayOftheWeek,
           "isOneTime": (request.body.isOneTime == null) ? "false" : request.body.isOneTime,
-          "classDate": (request.body.classDate == null) ? request.body.startTime : request.body.classDate
+          classDate: (request.body.classDate == null) ? request.body.startTime : request.body.classDate
```
Comment (zachsanchez113) [inline API/routes/api/make_schedule.js:90]: ?

File: API/routes/api/make_schedule.js [hunk 7]
```diff
@@ -110,6 +107,130 @@ router.route('/')
...
+    if (request.body.classDate) newAttr.classDate = request.body.classDate;
+    if (request.body.dayOftheWeek) newAttr.dayOftheWeek = request.body.dayOftheWeek;
+
+    /* Find the class in the DB and update it*/
+
+    scheduleModel.findOneAndUpdate({"className": request.body.className}, 'className startTime endTime isOneTime dayOftheWeek classDate', (e, query) => {
+      
+      if (e) return next(e);
+
+      if (!query[0] || query[0] == "") return next(createError(400, "No class exists to update"));
```
Comment (zachsanchez113) [inline API/routes/api/make_schedule.js:169]: It's just `query` when you use `findOneAndUpdate`, I went ahead and fixed that for you

File: API/routes/api/make_schedule.js [hunk 8]
```diff
@@ -110,6 +107,130 @@ router.route('/')
...
+
+      if (process.env.NODE_ENV == "DEBUG") {
+        
+        console.log("The list of new attributes: " + JSON.stringify(newAttr));
+        console.log("What was actually returned: " + JSON.stringify(query));
+
+      }
+
+      if (!query || query == "") return next(createError(400, "That class doesn't exist"));
```
Comment (zachsanchez113) [inline API/routes/api/make_schedule.js:178]: Wait wtf

File: API/routes/api/make_schedule.js [hunk 9]
```diff
@@ -110,6 +107,130 @@ router.route('/')
...
+    scheduleModel.delete({ className: deleteClass }, function (err) {
+
+      if (err) return next(createError(401, "Unable to delete the specified class."));
+      console.log(deleteClass + " was successfully deleted!");
+
+    });
+
+  });
+
+router.route('/')
```
Comment (zachsanchez113) [inline API/routes/api/make_schedule.js:233]: For future reference, the extra `router.route` is redundant

File: API/lib/schedule_db.js (hunk 4 above)
Comment (pcockrell6) [reply API/lib/schedule_db.js:77]: Commented it out for now.

File: API/routes/api/make_schedule.js (hunk 6 above)
Comment (pcockrell6) [reply API/routes/api/make_schedule.js:90]: Fixed; pushing again soon

File: API/routes/api/make_schedule.js (hunk 9 above)
Comment (pcockrell6) [reply API/routes/api/make_schedule.js:233]: Ok. For whatever reason, the code was upset about the other functions besides the first one attached to the router.route. I'll double check that the code is no longer upset by it somehow

**Pull request merged**